internal/util/util.go

```diff
@@ -60,15 +62,35 @@ func GetTempDir() string {
 }
 
 // ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
-func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+func ExportFromReader(input io.ReadCloser, store storage.Store, opts define.BuildOutputOption) error {
 	var err error
 	if !filepath.IsAbs(opts.Path) {
 		opts.Path, err = filepath.Abs(opts.Path)
 		if err != nil {
 			return err
 		}
 	}
-	if opts.IsDir {
+	if opts.Type == define.Image {
+		if opts.Attrs["push"] != "true" {
+			return nil
+		}
+
+		image := opts.Attrs["name"]
+		destSpec := opts.Attrs["name"]
+		dest, err := util.StringToImageReference(destSpec)
```

Comment:
I would much prefer for this to be called higher in the call stack, at the very least in `GetBuildOutput`, or possibly outside of `imagebuildah` at all (but that would be a larger change).

(And that would require a Go field instead of a `Attrs` only carrying strings.)